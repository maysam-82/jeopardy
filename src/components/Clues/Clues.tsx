import React from 'react';
import { IClue } from '../../types/category';
import { IStoreState } from '../../redux/reducers/index';
import { connect } from 'react-redux';
import Clue from '../Clue/Clue';

export interface ICluesProps {
	clues: IClue[];
	title: string;
}

export const Clues: React.SFC<ICluesProps> = ({ clues, title }) => {
	return (
		<div className="clues">
			<h4 className="text-center font-weight-bold">{title}</h4>
			<div className="clues-list">
				{clues.map(({ value, answer, question, id }) => (
					<Clue value={value} answer={answer} question={question} key={id} />
				))}
			</div>
		</div>
	);
};

const mapStateToProps = (state: IStoreState) => ({
	clues: state.category.clues,
});

export default connect(mapStateToProps)(Clues);
