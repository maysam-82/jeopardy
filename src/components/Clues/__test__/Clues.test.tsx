import React from 'react';
import { shallow } from 'enzyme';
import { Clues, ICluesProps } from '../Clues';
import { testClues } from '../../../fixtures/testData';

const testCluesProps: ICluesProps = {
	clues: testClues,
	title: 'test title',
};

describe('Clues component', () => {
	const component = shallow(<Clues {...testCluesProps} />);
	it('should render Clues component', () => {
		expect(component).toMatchSnapshot();
	});
	it('should render 2 `Clue` component', () => {
		expect(component.find('Clue').length).toEqual(2);
	});
	it('should render `title` equals to  `test title`', () => {
		expect(component.find('h4').text()).toEqual(testCluesProps.title);
	});
});
