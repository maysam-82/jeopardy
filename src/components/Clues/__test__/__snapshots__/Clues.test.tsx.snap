// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clues component should render Clues component 1`] = `
<div
  className="clues"
>
  <h4
    className="text-center font-weight-bold"
  >
    test title
  </h4>
  <div
    className="clues-list"
  >
    <Clue
      answer="chores"
      key="87819"
      question="They're routine tasks done around the house, & before you go out to play, you'd better do yours"
      value={200}
    />
    <Clue
      answer="charcoal"
      key="87825"
      question="It's the form of carbon your dad probably uses when he fires up the barbecue"
      value={400}
    />
  </div>
</div>
`;
